compiler/src/dotty/tools/dotc/transform/PruneErasedDefs.scala

```diff
@@ -38,7 +38,7 @@ class PruneErasedDefs extends MiniPhase with SymTransformer { thisTransform =>
     else tree
 
   override def transformValDef(tree: ValDef)(implicit ctx: Context): Tree =
-    if (tree.symbol.isEffectivelyErased && !tree.rhs.isEmpty)
+    if (tree.symbol.isEffectivelyErased && !tree.rhs.isEmpty && !tree.symbol.is(Inline, butNot = Method))
```

Comment:
It seems we already test ` !tree.symbol.is(Inline, butNot = Method)` in `isEffectivelyErased `

Reply:
This is an inline val that is `isEffectivelyErased`. Not all inline vals are erased, we need to retain it if it overrides or implements it. But we do not want to replace the body with `???` (it would fail Ycheck).

Reply:
Maybe the check could just be
```scala
    if (tree.symbol.is(Erased, butNot = Inline) && !tree.rhs.isEmpty)
```
I will try it.

Reply:
I use a more systematic approach. When pruning erased trees, if the tree is a constant we just keep it instead of replacing it with `???`.